src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/TensorPrimitives.netcore.cs

```diff
@@ -3098,6 +3332,802 @@ public static Vector128<float> Invoke(Vector128<float> x, Vector128<float> y)
 #endif
         }
 
+        private interface IIndexOfOperator
+        {
+            static abstract int Invoke(ReadOnlySpan<float> result);
+            static abstract int Invoke(Vector128<float> result, Vector128<int> resultIndex);
+            static abstract void Invoke(ref Vector128<float> result, Vector128<float> current, ref Vector128<int> resultIndex, Vector128<int> curIndex);
+            static abstract void Invoke(ref Vector128<float> result, Vector128<float> current, ref Vector128<int> resultIndex, Vector128<int> curIndex, Vector128<float> remainderMask);
+            static abstract int Invoke(Vector256<float> result, Vector256<int> resultIndex);
+            static abstract void Invoke(ref Vector256<float> result, Vector256<float> current, ref Vector256<int> resultIndex, Vector256<int> curIndex);
+            static abstract void Invoke(ref Vector256<float> result, Vector256<float> current, ref Vector256<int> resultIndex, Vector256<int> curIndex, Vector256<float> remainderMask);
+#if NET8_0_OR_GREATER
+            static abstract int Invoke(Vector512<float> result, Vector512<int> resultIndex);
+            static abstract void Invoke(ref Vector512<float> result, Vector512<float> current, ref Vector512<int> resultIndex, Vector512<int> curIndex);
+            static abstract void Invoke(ref Vector512<float> result, Vector512<float> current, ref Vector512<int> resultIndex, Vector512<int> curIndex, Vector512<float> remainderMask);
+#endif
+
+            static virtual float IdentityValue => throw new NotSupportedException();
+        }
+
+        /// <summary>MathF.Max(x, y) (but NaNs may not be propagated)</summary>
+        private readonly struct IndexOfMaxOperator : IIndexOfOperator
+        {
+            [MethodImpl(MethodImplOptions.AggressiveInlining)]
+            public static int Invoke(ReadOnlySpan<float> result)
+            {
+                float curMax = result[0];
+                int curIn = 0;
+                if (float.IsNaN(curMax))
+                {
+                    return curIn;
+                }
+
+                for (int i = 1; i < result.Length; i++)
+                {
+                    float current = result[i];
+                    if (float.IsNaN(current))
+                    {
+                        return i;
+                    }
+
+                    if (curMax == current)
+                    {
+                        if (IsNegative(curMax) && !IsNegative(current))
+                        {
+                            curMax = current;
+                            curIn = i;
+                        }
+                    }
+                    else if (current > curMax)
+                    {
+                        curMax = current;
+                        curIn = i;
+                    }
+                }
+
+                return curIn;
+            }
```

Comment:
If you keep a separate scalar function, you should be able to then call it from both the netcore and netstandard implementations.

Reply:
Good idea. I'll do that.